### ISSUE TITLE
How do I query graph status in Apollo Studio?

### ISSUE THREAD

**Issue by jeantessier** [apollographql/apollo-tooling#2031]
## Problem Definition

TL;DR: How do we know the graph is good after updating multiple services with `service:push`?

We have a modular monolith where each module exposes its own portion of a federated schema as a if it were an independent service.  We use a GraphQL gateway in front to provide a single access point to the frontend.  During deployments of the monolith by our CI/CD, we have to call `service:push` for each _service_.

Sometimes, a service with an extended type may get pushed before the service with the base type, resulting in a temporary composition error in Apollo Studio.

Let me illustrate with a contrived example.

Schema for service A:
```
type Account @key(fields: "accountId") {
    accountId: ID!
    user: User!
    # Other account-specific fields
}

extend type User @key(fields: "userId") {
    userId: ID! @external
    accounts: [Account!]!
}
```

Schema for service B:
```
type User @key(fields: "userId") {
    userId: ID!
    name: String!
}
```

We have a script that calls `service:push` for each service.  For each call, if it modifies the federated graph in Apollo Studio and the graph contains errors by the time the call is done, the script fails and returns non-zero.  If the specific service hasn't changed since the last time it was pushed, the call exits with a zero status, even though the graph in Apollo Studio may still contain errors.

We don't want to ring alarms for transient errors.  Only if the graph still has errors after we've applied all the pushes.

## Requested Feature

We need a way to determine the health of the federated graph without having to push a change to it.

One option would be for `service:push` to exit with a non-zero status any time the graph contains errors.  This way, we could call `service:push` for each service and only retain the exit status of the last call.

Another option would be to have a another command dedicated to returning the health of the federated schema.  This way, we could call `service:push` for each service and then fetch the resulting graph health.
